packages/node_modules/@webex/plugin-meetings/test/unit/spec/meetings/index.js

```diff
@@ -0,0 +1,321 @@
+              MeetingCollection.getByKey = sinon.stub().returns(true);
+            });
+            describe('#getMeetingByType', () => {
+              it('gets the Meeting instance from MeetingCollection using type and value', () => {
+                staticMeetings.getMeetingByType('test', 'test');
+                assert.calledOnce(MeetingCollection.getByKey);
+                assert.calledWith(MeetingCollection.getByKey, 'test', 'test');
+              });
+            });
+          });
+          describe('MeetingCollection getAll proxies', () => {
+            beforeEach(() => {
+              MeetingCollection.getAll = sinon.stub().returns(true);
+            });
+            describe('#getAllMeetings', () => {
+              it('calls MeetingCollection to get all meetings with supplied options', () => {
+                staticMeetings.getAllMeetings({test: 'test'});
+                assert.calledOnce(MeetingCollection.getAll);
+                assert.calledWith(MeetingCollection.getAll, {test: 'test'});
+              });
+            });
+            describe('#getActiveMeetings', () => {
+              it('calls MeetingCollection to get all active meetings using the active option', () => {
+                staticMeetings.getActiveMeetings();
+                assert.calledOnce(MeetingCollection.getAll);
+                assert.calledWith(MeetingCollection.getAll, {active: true});
+              });
+            });
+            describe('#getScheduledMeetings', () => {
+              it('calls MeetingCollection to get all scheduled meetings using the scheduled option', () => {
+                staticMeetings.getScheduledMeetings();
+                assert.calledOnce(MeetingCollection.getAll);
+                assert.calledWith(MeetingCollection.getAll, {scheduled: true});
+              });
+            });
+          });
+        });
+      });
+      describe('#syncMeetings', () => {
+        describe('succesful requests', () => {
+          beforeEach(() => {
+            spark.meetings.request.getActiveMeetings = sinon.stub().returns(Promise.resolve({loci: [{url: 'test'}]}));
+          });
+          describe('when meeting is returned', () => {
+            let onFullLocus;
+            beforeEach(() => {
+              onFullLocus = sinon.stub().returns(true);
+              MeetingCollection.getByKey = sinon.stub().returns({locusInfo: {onFullLocus}});
+            });
+            it('tests the sync meeting calls for existing meeting', async () => {
+              await spark.meetings.syncMeetings();
+              assert.calledOnce(spark.meetings.request.getActiveMeetings);
+              assert.calledOnce(MeetingCollection.getByKey);
+              assert.calledOnce(onFullLocus);
+              assert.calledWith(spark.meetings.request.getActiveMeetings);
+              assert.calledWith(MeetingCollection.getByKey, 'locusUrl', 'test');
+              assert.calledWith(onFullLocus, {url: 'test'});
```

Comment:
I don't think we want to assert on individual internal methods being called. We should be able to change those and as long as the end result is the same, the test should pass.

Reply:
hmmmmm, i agree with you, but we also want to make sure when we make changes, that the function is doing everything it was initially supposed to, so if somebody changes something internal, that was initially meant to also be called (for example, unsetlocalStream AND unset localShare), we would have broken behavior.